coq 8.5pl1

lukenels opened the pull request:
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Does your submission pass `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Have you built your formula locally prior to submission with `brew install <formula>`?

ilovezfs commented on Formula/coq.rb:
```diff
@@ -48,4 +48,10 @@ def caveats; <<-EOS.undent
+  test do
+    unless File.exist? "#{bin}/coqc" then return false end
+    system("#{bin}/coqc", "-v")
+    $?.exitstatus == 0
```
I'd like to see a test here that does something with a proof.

mikemcquaid replied:
Also, `system` already verified the exit code.

DomT4 replied:
Also if `coqc` doesn't exist and you call it in the test it fails with a pretty obvious error anyway, we don't need to throw a manual error ourselves there.

DomT4 replied:
He says, scrolling down the page to realise Mike said the same thing 20 minutes ago. Apologies 😄.

mikemcquaid commented on Formula/coq.rb:
```diff
@@ -48,4 +48,10 @@ def caveats; <<-EOS.undent
   end
+
+  test do
+    unless File.exist? "#{bin}/coqc" then return false end
```
You don't need to test this.

lukenels commented:
Thanks for the reviews everyone. I've updated the diff to only contain the call to system. If people feel there's a need, I could add better testing, except that:

1. The build process itself uses the Coq compiler to check all of the proof in the standard library; if the compiler was broken the build would have failed.
2. As far as I am aware there is no test suite for Coq, the best I could find was on the [Coq Installation](https://coq.inria.fr/cocorico/Installation%20of%20Coq%20on%20Linux) page, which describes only the rudimentary testing I've implemented here.

I realize that having a good test suite is important for formulae, so if anyone is aware of a better Coq test suite or feels strongly about the creation of one, then that can be looked into. Otherwise, we're no worse off than we have been on the previous versions of the Coq formula.

apjanke commented:
We don't need to do a full test suite, just a "smoke test" (we like to keep the tests quick, as they're used a lot in the CI build process). But for compiler tools and analyzers, the equivalent of compiling or checking a "Hello World" is nice. It's good to have this in the test, and not just in the build process, so that it can catch breakage that happens to the formula *after* installation, such as dylibs or other files it depends on being changed.

Could you add the equivalent of checking a "Hello World" proof as a test?

lukenels commented:
@apjanke: sounds good, I will work on that and have an updated diff tomorrow at the latest.

lukenels commented:
I've updated the test to read and typecheck a simple coq file that defines natural numbers, addition, and proves the O as the right additive identity.

apjanke commented:
Perfect. I'll merge this when the CI build finishes.

mikemcquaid commented:
@lukenels Thanks for the great test. It makes it much easier for us to see regressions in our packaging 👏

mikemcquaid commented on Formula/coq.rb:
```diff
@@ -13,8 +13,8 @@ def message; <<-EOS.undent
   homepage "https://coq.inria.fr/"
-  url "https://coq.inria.fr/distrib/V8.5/files/coq-8.5.tar.gz"
-  sha256 "89a92fb8b91e7cb0797d41c87cd13e4b63bee76c32a6dcc3d7c8055ca6a9ae3d"
+  url "https://coq.inria.fr/distrib/8.5pl1/files/coq-8.5pl1.tar.gz"
```
This version isn't being detected correctly. unfortunately. Please add a dedicated `version "8.5pl1"` line. Thanks!

lukenels replied:
Just added it

UniqMartin commented:
Merged in 019d8048e9b28c045d118fe3686fb0acc4b91947. Thank you for this contribution to Homebrew and for a very nice test, @lukenels! :tada:
